packages/mdc-fab/_mixins.scss

```diff
@@ -0,0 +1,129 @@
+@import "@material/ripple/mixins";
+@import "@material/theme/mixins";
+
+@mixin mdc-fab-base {
+  @include mdc-ripple-base;
+
+  display: inline-flex;
+  position: relative;
+  justify-content: center;
+  width: 56px;
+  height: 56px;
+  padding: 0;
+  transition: box-shadow 280ms $mdc-animation-standard-curve-timing-function;
+  border: none;
+  border-radius: 50%;
+  cursor: pointer;
+  user-select: none;
+  box-sizing: border-box;
+  fill: currentColor;
+  -moz-appearance: none;
+  -webkit-appearance: none;
+  overflow: hidden;
+
+  @include mdc-elevation(6);
+
+  &:active {
+    @include mdc-elevation(12);
+  }
+
+  &:active,
+  &:focus {
+    // TODO(acdvorak): Should this be paired with states and/or ripple? We don't want to disable outline
+    // (an accessibility/usability feature) unless we're confident that there is also a visual indication that the
+    // element has focus. If the client has customized the DOM in some unexpected way, and is certain that another
+    // element will receive focus instead, they can always override this property manually in their CSS.
+    outline: none;
+  }
+
+  &:hover {
+    cursor: pointer;
+  }
+
+  &::-moz-focus-inner {
+    padding: 0;
+    border: 0;
+  }
+
+  // This allows for using SVGs within them to align properly in all browsers.
+  // Can remove once: https://bugzilla.mozilla.org/show_bug.cgi?id=1294515 is resolved.
+
+  // stylelint-disable selector-max-type
+  // postcss-bem-linter: ignore
+  > svg {
+    width: 100%;
+  }
+  // stylelint-enable selector-max-type
+}
+
+@mixin mdc-fab-mini {
+  width: 40px;
+  height: 40px;
+}
+
+@mixin mdc-fab-disabled {
+  cursor: default;
+  pointer-events: none;
+}
+
+//**
+// @param {color|ThemeProp} $fab-theme-config.bg-color
+// @param {color|ThemeProp} $fab-theme-config.fg-color
+// @param {RippleConfig} $fab-theme-config.ripple-config @see mdc-ripple-default-config_ in mdc-ripple/_mixins.scss.
+// @param {color|ThemeProp} $fab-theme-config.tap-highlight-color
+//*
+@mixin mdc-fab-theme($fab-theme-config) {
+  $bg-color: map-get($fab-theme-config, bg-color);
+  $fg-color: map-get($fab-theme-config, fg-color);
+  $ripple-config: map-get($fab-theme-config, ripple-config);
+  $tap-highlight-color: map-get($fab-theme-config, tap-highlight-color);
+
+  @if $ripple-config {
+    @include mdc-ripple-bg(map-merge((pseudo: "::before"), $ripple-config));
+    @include mdc-ripple-fg(map-merge((pseudo: "::after"), $ripple-config));
+  }
+
+  @if $tap-highlight-color {
+    @if $ripple-config {
+      &:not(.mdc-ripple-upgraded) {
+        @include mdc-theme-prop(-webkit-tap-highlight-color, $tap-highlight-color);
+      }
+    }
+
+    @else {
+      @include mdc-theme-prop(-webkit-tap-highlight-color, $tap-highlight-color);
+    }
+  }
+
+  @if $bg-color {
+    @include mdc-theme-prop(background-color, $bg-color);
+  }
+
+  @if $fg-color {
+    @include mdc-theme-prop(color, $fg-color);
+  }
+}
+
+@mixin mdc-fab-icon {
```

Comment:
As we discussed offline, we will keep the mixin, but rename it using BEM. We can revisit this decision later.